### PR TITLE
FOGL-2243 - Filters unit tests for internal defs

### PR DESCRIPTION
**Pull request by amarendra-dianomic**

### CODE REVIEW

File: tests/unit/python/foglamp/services/core/api/test_filters.py [hunk 1]
```diff
@@ -6,17 +6,18 @@
...
 import asyncio
 import json
-from unittest.mock import MagicMock, patch
+from unittest.mock import MagicMock, patch, call
 from aiohttp import web
 import pytest
 
 from foglamp.services.core import routes
 from foglamp.services.core import connect
+from foglamp.common.storage_client.payload_builder import PayloadBuilder
```
Comment (ashish-jabble) [inline tests/unit/python/foglamp/services/core/api/test_filters.py]: remove unused statement

File: tests/unit/python/foglamp/services/core/api/test_filters.py [hunk 2]
```diff
@@ -6,17 +6,18 @@
...
 from foglamp.common.storage_client.storage_client import StorageClientAsync
 from foglamp.common.storage_client.exceptions import StorageServerError
 from foglamp.services.core.api import filters
 from foglamp.services.core.api.filters import _LOGGER
-from foglamp.common.configuration_manager import ConfigurationManager
+from foglamp.common.configuration_manager import ConfigurationManager, ConfigurationCache
```
Comment (ashish-jabble) [inline tests/unit/python/foglamp/services/core/api/test_filters.py]: remove unused statement

File: tests/unit/python/foglamp/services/core/api/test_filters.py [hunk 3]
```diff
@@ -660,3 +661,246 @@ def q_result(*args):
...
+        assert mock_payload == p
+
+        calls_tbl = [call('filter_users', '{"where": {"column": "name", "condition": "=", "value": "scale1", "and": {"column": "user", "condition": "=", "value": "random1"}}}'),
+                     call('filter_users', '{"where": {"column": "name", "condition": "=", "value": "python35a", "and": {"column": "user", "condition": "=", "value": "random1"}}}'),
+                     call('filter_users', '{"where": {"column": "name", "condition": "=", "value": "meta1", "and": {"column": "user", "condition": "=", "value": "random1"}}}')]
+        delete_tbl_patch.assert_has_calls(calls_tbl, any_order=True)
+
+        calls_child = [call('random1', 'random1_scale1'),
+                        call('random1', 'random1_python35a'),
+                        call('random1', 'random1_meta1')]
```
Comment (ashish-jabble) [inline tests/unit/python/foglamp/services/core/api/test_filters.py]: Fix the identation

File: tests/unit/python/foglamp/services/core/api/test_filters.py [hunk 4]
```diff
@@ -660,3 +661,246 @@ def q_result(*args):
...
+        create_category_mock.assert_has_calls(calls_create_cat, any_order=True)
+
+        calls_create_child = [call(category_name='random1', children=['random1_scale1', 'random1_meta2'])]
+        create_child_category_mock.assert_has_calls(calls_create_child, any_order=True)
+
+        calls_update = [call('random1_meta2', {'assetName': 'test1'})]
+        update_config_bulk_mock.assert_has_calls(calls_update, any_order=True)
+
+        calls_insert = [call('filter_users', '{"user": "random1", "name": "meta2"}')]
+        insert_tbl_patch.assert_has_calls(calls_insert, any_order=True)
```
Comment (ashish-jabble) [inline tests/unit/python/foglamp/services/core/api/test_filters.py]: This is failing on mutiple time run at my local and also on CI as well on first commit, please fix the assertion

File: tests/unit/python/foglamp/services/core/api/test_filters.py [hunk 5]
```diff
@@ -660,3 +661,246 @@ def q_result(*args):
...
+        }
+        a, b = filters._delete_keys_from_dict(in_dict_del, lst_keys, deleted_values={}, parent=None)
+        assert out_dict_del, deleted_values == (a, b)
+
+    async def test_delete_child_filters(self, mocker):
+        # GIVEN
+        user_name_mock = 'random1'
+        new_list_mock = ['scale2', 'python35b', 'meta2']
+        old_list_mock = ['scale1', 'python35a', 'meta1']
+        in_dict_del = {
```
Comment (ashish-jabble) [inline tests/unit/python/foglamp/services/core/api/test_filters.py]: unused dict

File: tests/unit/python/foglamp/services/core/api/test_filters.py [hunk 6]
```diff
@@ -660,3 +661,246 @@ def q_result(*args):
...
+
+        # WHEN
+        await filters._delete_configuration_category(storage_client_mock, "test")
+
+        # THEN
+        args, kwargs = delete_tbl_patch.call_args
+        assert 'configuration' == args[0]
+        p = json.loads(args[1])
+        assert mock_payload == p
+        # cache_remove.assert_called_once_with("test")
```
Comment (ashish-jabble) [inline tests/unit/python/foglamp/services/core/api/test_filters.py]: Remove this commented line

File: tests/unit/python/foglamp/services/core/api/test_filters.py [hunk 7]
```diff
@@ -660,3 +661,246 @@ def q_result(*args):
...
+                 'value': 'test'
+               }
+        }
+        storage_client_mock = MagicMock(StorageClientAsync)
+        c_mgr = MagicMock(ConfigurationManager)
+
+        mock_connect = mocker.patch.object(connect, 'get_storage_async', return_value=storage_client_mock)
+        delete_tbl_patch = mocker.patch.object(storage_client_mock, 'delete_from_tbl', return_value=asyncio.sleep(.1))
+        cache_manager = mocker.patch.object(c_mgr, '_cacheManager')
+        cache_remove = mocker.patch.object(cache_manager, 'remove', return_value=MagicMock())
```
Comment (praveen-garg) [inline tests/unit/python/foglamp/services/core/api/test_filters.py]: is this being used?

Comment (amarendra-dianomic) [reply tests/unit/python/foglamp/services/core/api/test_filters.py]: This is a TODO item. Comment added.

**Pull request merged**